encoding/msgpack/msgpack.go

```diff
@@ -19,6 +22,15 @@ func init() {
 	CodecHandle.RecursiveEmptyCheck = true
 	CodecHandle.WriteExt = true
 	CodecHandle.PositiveIntUnsigned = true
+
+	LenientCodecHandle = new(codec.MsgpackHandle)
+	// allow unknown fields to ensure forward compatibility.
+	LenientCodecHandle.ErrorIfNoField = false
```

Comment:
Why do we want allow for unmatched field instead of updating the struct with field changes?

Reply:
The main case is probably when someone did not update the SDK. Suppose there are new fields in the response, we have the following options:
1) allow the SDK to continue working by ignoring the missing field.
2) force the application to update the SDK to a new version which has the missing field.

Reply:
We should also ensure the types are updated / correct.